andela/ah-robotics-frontend
# #162949029 Setting up redux

#### What does this PR do?

This PR implements react-redux using the `redux-saga` library.

#### Description of Task to be completed?

- [x] Implement Redux store
- [x] set up necessary files needed to create the store

#### How should this be manually tested?

1. Clone this repository on your local machine using 

     `git clone https://github.com/andela/ah-robotics-frontend.git`
2. Change your working directory to the project's directory using 
     
     `cd ah-robotics-frontend`
3. Change to the specific branch `ch-setup-redux-162949029` using 

     `git checkout ch-setup-redux-162949029`

4. Run `npm install` to install all the modules listed in the `package.json` file.

5. Run `npm start` command to start the application.

6. The application should start successfully displaying a page as shown below.

<img width="1433" alt="screenshot 2019-02-07 at 12 39 52" src="https://user-images.githubusercontent.com/6994982/52402505-80deda80-2ad5-11e9-8f19-35130eeb0619.png">

7. Click on the `add-button` it should increment the count as displayed above.

8. Click on the `Redux DevTools` extension on the Chrome browser and it should display the store as shown below.

> The store will update and it will display the count number

> Ps. If you **do not** have `Redux DevTools` installed, find the instructions listed below on how to install the extension.

<img width="714" alt="screenshot 2019-02-06 at 18 53 29" src="https://user-images.githubusercontent.com/6994982/52353907-863e1580-2a40-11e9-931a-8820e3ab8b5f.png">


#### Additional steps to set up your browser for testing React Apps better.

> These tools are not necessary to run the application, they are just extra tools to help you test and debug React applications quickly

1. Install React DevTools extension for Chrome

> Find the extension [Here](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi)

2. Install Redux extension for Chrome

>Find the extension [Here](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en)

#### What are the relevant pivotal tracker stories?

[#162949029](https://www.pivotaltracker.com/story/show/162949029)
